### PR TITLE
Handle FormServ field_options object in form control

### PR DESCRIPTION
**Pull request by srivalli-sivaramasankaran**
- updated test cases for form and form control

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] My commits have standard messages as mentioned in [Contributing Guidelines](./../blob/next/CONTRIBUTING.md)

### CODE REVIEW

File: packages/crayons-core/src/components/form-control/form-control.tsx [hunk 1]
```diff
@@ -253,7 +253,14 @@ export class FormControl {
             ...controlProps,
             options: this.choices,
           };
-
+          // This is to handle formserv payload which might contain a field_options object, that has parameters, option_value_path and option_label_path,
+          // that denotes which property of choices object(form schema) needs to be displayed as label and which should be stored in the backend as value
+          if (this.fieldProps?.field_options?.option_value_path)
```
Comment (arvindanta) [inline packages/crayons-core/src/components/form-control/form-control.tsx:258]: can you extract `this.fieldProps?.field_options?` to a variable and use it.

Comment (srivalli-sivaramasankaran) [reply packages/crayons-core/src/components/form-control/form-control.tsx]: done.

File: packages/crayons-core/src/components/form/readme.md [hunk 2]
```diff
@@ -291,6 +291,72 @@ fields: [ // Each item in this array corresponds to a crayons input component.
...
 ```
 
+## Text and Value indicators for select component choices
+
+The choices for select component(dropdown/multi_select) can follow any of the below formats.
+
+1. Using field_options object. Pass the key name that needs to be displayed as the dropdown option's text using, 'option_label_path' and the key name that needs to to be processed in the backend using, 'option_value_path'.
+
+```
+{
```
Comment (arvindanta) [inline packages/crayons-core/src/components/form/readme.md:301]: can you add a working example for using `field_options` with `option label and value path` ?

**Review by arvindanta — CHANGES_REQUESTED** [review]

File: packages/crayons-core/src/components/form/readme.md (hunk 2 above)
Comment (srivalli-sivaramasankaran) [reply packages/crayons-core/src/components/form/readme.md:301]: added

**Review by arvindanta — APPROVED** [review]

**Pull request merged**